### DISCUSSION TITLE
"groupby" choose "orbital" or "spin"  in nonlinear calculations ?

### DISCUSSION THREAD

**Discussion by congh-m** [mailhexu/TB2J#27] [Q&A] [answered]
When we run wann2J.py in nonlinear calculations,  is there any difference if the "groupby" option is set to "orbital" or "spin"?  How do we decide which one to choose?

**Comment by mailhexu** [comment, marked as answer]
Hi,
Yes this makes a huge difference. The "groupby" option specifies the order
of the Wannier functions, which depends on the DFT code (and sometimes the
version of the DFT code too). In principle, this should be documented.  You
can also get some hint from the Wannier center and spread to guess the
order.

You can find the description of the option at:
https://tb2j.readthedocs.io/en/latest/src/wannier.html?highlight=groupby#noncollinear-calculation

For calculations with non-collinear spin, the –spinor option should be
used. It is also necessary to specify whether in the Hamiltonian the order
of the basis, either group by spin (orb1_up, orb2_up, … orb1_down,
orb2_down, …) or by orbital (orb1_up, orb1_down, orb2_up, orb2_down,…),
with the –groupby option (either spin or orbital).

Best wishes,
HeXu
